### PR TITLE
feature: Optimize Fillwidthitem Image for Mobile

### PR DESCRIPTION
**Pull request by icirellik**
This change optimizes the Fillwidthitem images for mobile and updates it
to use the new resizer functionality.

### CODE REVIEW

**Comment by dzucconi** [comment]
I don't follow why we would want to do this. These should be pixel densities which aren't going to change. We shouldn't be passing custom ones but rather serving correct srcSets and letting the device choose. The images should be scaled exactly to the dimensions they are rendered at and if we want to dynamically adjust the quality then do so with a quality param.

**Comment by icirellik** [comment]
@dzucconi I'm gathering some screenshots containing the amount of data being sent over the wire. This change does not affect how desktop images are rendered however the images we are sending to mobile are quite a bit larger than they need to be.

**Comment by icirellik** [comment]
@dzucconi I've updated it to use quality as we discussed and will plug in the final values once we have them!

File: src/v2/Utils/resized.ts [hunk 1]
```diff
@@ -9,12 +9,38 @@ type Resize = Parameters<typeof resize>
...
+  }
 ): Sized => {
-  const _1x = resize(src, { width, height, ...rest })
+  let dpi1x
+  let dpi2x
+  if (typeof quality === "number") {
+    dpi1x = dpi2x = quality
+  } else {
+    dpi1x = quality[0]
+    dpi2x = quality[1]
```
Comment (dzucconi) [inline src/v2/Utils/resized.ts:31]: Could we rename these quality1x/quality2x or something?

**Comment by dzucconi** [comment]
For later reference: here's the codepen we used to taste test different quality/resolutions: https://codepen.io/dzuc/pen/NWdVOaa

File: src/v2/Components/Artwork/FillwidthItem.tsx [hunk 2]
```diff
@@ -130,11 +107,16 @@ export class FillwidthItemContainer extends React.Component<
       return null
     }
 
+    const scaledImage = resized(artwork.image.url, {
+      resizeTo: artwork.image.aspectRatio === 1 ? "fill" : "fit",
```
Comment (dzucconi) [inline src/v2/Components/Artwork/FillwidthItem.tsx:111]: Ah so: there's a separate function for crop in the same file there. It'd be clearer to just leave those alone and swap out the functions I think.

File: src/v2/Utils/resized.ts [hunk 3]
```diff
@@ -32,10 +57,58 @@ type Crop = Parameters<typeof crop>
...
+    ...rest,
+  })
+
+  return {
+    src: _1x,
+    srcSet: `${_1x} 1x, ${_2x} 2x`,
+  }
+}
+
+export const geminiImage = ({
```
Comment (dzucconi) [inline src/v2/Utils/resized.ts:88]: I'm fine with this but what's wrong with

```ts
const fn = artwork.image.aspectRatio === 1 ? crop : resize
```

**Review by mc-jones — APPROVED** [review]

**Review by dzucconi — APPROVED** [review]

File: src/v2/Components/Artwork/FillwidthItem.tsx [hunk 4]
```diff
@@ -130,11 +103,25 @@ export class FillwidthItemContainer extends React.Component<
...
+          height: this.imageHeight,
+        })
+        break
+      case "fill":
+      default:
+        scaledImage = cropped(artwork.image.url, {
+          width: this.imageWidth,
+          height: this.imageHeight,
+        })
+    }
```
Comment (icirellik) [inline src/v2/Components/Artwork/FillwidthItem.tsx:120]:
```suggestion
    const transform = artwork.image.aspectRatio === 1 ? cropped : resized
    const scaledImage = transform(artwork.image.url, {
      width: this.imageWidth,
      height: this.imageHeight,
    })
```

**Pull request merged**